lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)?
+# Space Complexity: O(n) ?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  split_string_array = string.downcase.split("")
+  words_array = split_string_array.select{|letter| letter[/[a-z]/]}
+
+  reverse_words_array = []
+  x = 0
+  length = words_array.length
+  until words_array[x] == nil
+    reverse_words_array << words_array[length -(x+1)]
+    x += 1
+  end
+
+  return reverse_words_array == words_array ? true : false
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)?
+# Space Complexity: 0(n)?
 def longest_prefix(strings)
```

Comment:
This one isn't working, you need 2 loops here.

1. Outer loop which goes through all the characters of at least one word
2. Inner loop which goes through each word
3.  If statement in the inner loop, if the current character matches the same character in the first word, continue, otherwise return the longest prefix so far.
4.  Outside the inner loop add the current letter to the longest prefix.